analysis_notebooks/Geography Analysis.ipynb

```diff
@@ -306,154 +311,253 @@
+    "            try:\n",
+    "                iso3_to_scaledpop[iso3] = count / CountryInfo(country.replace('+', ' ')).population()\n",
+    "            except KeyError:\n",
+    "                pass\n",
+    "    #             print(\"{} not found in CountryInfo\".format(country))\n",
     "        try:\n",
-    "            iso3_to_scaledpop[iso3] = count / CountryInfo(country.replace('+', ' ')).population()\n",
+    "            subregion = iso3_to_subregion[iso3]\n",
+    "            if subregion in subregion_counts.keys():\n",
+    "                subregion_counts[subregion] += count\n",
+    "            else:\n",
+    "                subregion_counts[subregion] = count\n",
     "        except KeyError:\n",
-    "            pass\n",
-    "#             print(\"{} not found in CountryInfo\".format(country))\n",
-    "    try:\n",
-    "        subregion = iso3_to_subregion[iso3]\n",
-    "        if subregion in subregion_counts.keys():\n",
-    "            subregion_counts[subregion] += count\n",
-    "        else:\n",
-    "            subregion_counts[subregion] = count\n",
-    "    except KeyError:\n",
-    "        print(\"This country's subregion not found: {}\".format(country))\n",
+    "            print(\"This country's subregion not found: {}\".format(country))\n",
+    "    for key in iso3_to_scaledpop.keys():\n",
+    "        iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
     "\n",
-    "for key in iso3_to_scaledpop.keys():\n",
-    "    iso3_to_scaledpop[key] /= min(iso3_to_scaledpop.values())\n",
+    "\n",
+    "else:\n",
+    "    counts_gps = pickle.load(open(\"results/{}/geo_ctr_gps.pkl\".format(folder_name), \"rb\"))\n",
+    "    region_to_id = counts_gps[\"region_to_id\"]\n",
+    "    id_to_gps = counts_gps[\"id_to_gps\"]\n",
+    "    geo_boundaries = dataset.geo_boundaries\n",
+    "    choro_data = pd.read_csv(dataset.choropleth_filepath)\n",
+    "\n",
+    "    counts = {}\n",
+    "    for region in region_to_id:\n",
+    "        counts[region] = len(region_to_id[region])\n",
+    "    total = sum(counts.values())\n",
+    "    country_count_phrases = []\n",
+    "\n",
+    "\n",
+    "    for country, count in sorted(counts.items(), key=lambda x: x[1], reverse=True):\n",
+    "        country_count_phrases.append(\"{0}: {1}   {2}%\".format(country, count, round(100.*count/total)))\n",
+    "\n",
+    "    colors = [\n",
+    "        'red',\n",
+    "        'blue',\n",
+    "        'gray',\n",
+    "        'darkred',\n",
+    "        'beige',\n",
+    "        'green',\n",
+    "        'purple',\n",
+    "        'pink',\n",
+    "        'green',\n",
+    "        'purple',\n",
+    "        'pink',\n",
+    "    ]\n",
+    "    subregion_counts = {}\n",
+    "    for key in region_to_id:\n",
+    "        filename_arr = region_to_id[key]\n",
+    "        subregion_counts[key] = len(filename_arr)\n",
     "\n",
     "def country_counts_num(topn):\n",
```

Comment:
what does this do for the gps coordinates? also, maybe rename function as well as change print's to not overfit to "country"?

Reply:
renamed function and print statements to "region". 
subregion_counts for gps removed for now. will add in next pr as discussed